### PR TITLE
Vctr 36 implement front, back, data (with vector bool solution)

### LINKED ISSUE #17: Implement front, back, data (with vector bool solution)

### PR DESCRIPTION
**Pull request by JanosGit**

### CODE REVIEW

File: include/vctr/Containers/Vector.h [hunk 1]
```diff
@@ -50,18 +70,24 @@ struct DefaultVectorAllocator<ElementType> : AlignedAllocator<ElementType, Confi
     Vector<float, std::allocator> v;
     @endcode
 
+    There is one special case with Vector<bool>: For this particular case we use don't use
```
Comment (iromur) [inline include/vctr/Containers/Vector.h]: `use` or `don't use`? 😆

File: include/vctr/Containers/VectorBoolWorkaround.h [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+*/
+
+#pragma once
+
+namespace vctr::detail
+{
+
+/** An internal helper class to get around the std::vector<bool> specialisation limits.
+
+    std::vector<bool> behaves different to the default implementation as it acts like a
```
Comment (iromur) [inline include/vctr/Containers/VectorBoolWorkaround.h]: nitpicking: `behaves differently` or `differs from the default…`

File: include/vctr/Containers/VectorBoolWorkaround.h [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+private:
+    using StdVectorType = std::vector<std::byte, Allocator>;
+
+    static bool&       toBool (std::byte& element)       { return *reinterpret_cast<bool*> (&element); }
+    static const bool& toBool (const std::byte& element) { return *reinterpret_cast<const bool*> (&element); }
+    static bool*       toBool (std::byte* ptr)           { return reinterpret_cast<bool*> (ptr); }
+    static const bool* toBool (const std::byte* ptr)     { return reinterpret_cast<const bool*> (ptr); }
+
+public:
+
```
Comment (iromur) [inline include/vctr/Containers/VectorBoolWorkaround.h]: superfluous newline?

**Review by iromur — APPROVED** [review]
Looks awesome!
Happy New Year!

File: include/vctr/Containers/Vector.h (hunk 1 above)
Comment (JanosGit) [reply include/vctr/Containers/Vector.h]: 🙈

**Pull request merged**